clippy_lints/src/formatting.rs

```diff
@@ -146,44 +147,40 @@ fn check_assign(cx: &EarlyContext<'_>, expr: &ast::Expr) {
 
 /// Implementation of the `SUSPICIOUS_ELSE_FORMATTING` lint for weird `else`.
 fn check_else(cx: &EarlyContext<'_>, expr: &ast::Expr) {
-    if let Some((then, &Some(ref else_))) = unsugar_if(expr) {
-        if (is_block(else_) || unsugar_if(else_).is_some())
-            && !differing_macro_contexts(then.span, else_.span)
-            && !in_macro(then.span)
-            && !in_external_macro(cx.sess, expr.span)
-        {
-            // workaround for rust-lang/rust#43081
-            if expr.span.lo().0 == 0 && expr.span.hi().0 == 0 {
-                return;
-            }
+    if_chain! {
+        if let Some((then, &Some(ref else_))) = unsugar_if(expr);
+        if is_block(else_) || unsugar_if(else_).is_some();
+        if !differing_macro_contexts(then.span, else_.span);
+        if !in_macro(then.span) && !in_external_macro(cx.sess, expr.span);
 
-            // this will be a span from the closing ‘}’ of the “then” block (excluding) to
-            // the
-            // “if” of the “else if” block (excluding)
-            let else_span = then.span.between(else_.span);
+        // workaround for rust-lang/rust#43081
+        if expr.span.lo().0 != 0 && expr.span.hi().0 != 0;
 
-            // the snippet should look like " else \n    " with maybe comments anywhere
-            // it’s bad when there is a ‘\n’ after the “else”
-            if let Some(else_snippet) = snippet_opt(cx, else_span) {
-                let else_pos = else_snippet.find("else").expect("there must be a `else` here");
+        // this will be a span from the closing ‘}’ of the “then” block (excluding) to
+        // the
+        // “if” of the “else if” block (excluding)
```

Comment:
NIT: formatting

Is it possible to add a test for this ICE? Otherwise r=me with formatting fixed.

Reply:
nit fixed =)

I tried since yesterday but couldn't come up with a test case without external dependencies on relm and gtk. I think it's fine to merge it in this case since the ICE was triggered by an explicit `expect` in our code and it would only regress if we were to add it back.